### ISSUE TITLE
google_maps_flutter crashes on iPad 12.5.7 on brand new test app

### ISSUE THREAD

**Issue by christian-froh** [flutter/flutter#138325]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Create a flutter app and add google_maps_flutter: ^2.5.0 to it

or just get this test app https://github.com/christian-froh/flutter_google_maps_ipad_crash

### Expected results

App opens

### Actual results

App crashes on startup

### Code sample

https://github.com/christian-froh/flutter_google_maps_ipad_crash

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

<img width="1437" alt="Bildschirmfoto 2023-11-12 um 19 09 31" src="https://github.com/flutter/flutter/assets/409788/6b285781-f442-4c69-aadd-29e5afb7a4ee">

</details>


### Logs

<details><summary>Logs</summary>

```console
dyld`:
    0x102221000 <+0>:   mov    x28, sp
    0x102221004 <+4>:   and    sp, x28, #0xfffffffffffffff0
    0x102221008 <+8>:   mov    x0, #0x0
    0x10222100c <+12>:  mov    x1, #0x0
    0x102221010 <+16>:  stp    x1, x0, [sp, #-0x10]!
    0x102221014 <+20>:  mov    x29, sp
    0x102221018 <+24>:  sub    sp, sp, #0x10
    0x10222101c <+28>:  ldr    x0, [x28]
    0x102221020 <+32>:  ldr    x1, [x28, #0x8]
    0x102221024 <+36>:  add    x2, x28, #0x10
    0x102221028 <+40>:  adrp   x4, -1
    0x10222102c <+44>:  add    x4, x4, #0x0
    0x102221030 <+48>:  adrp   x3, 90
    0x102221034 <+52>:  ldr    x3, [x3, #0xad8]
    0x102221038 <+56>:  sub    x3, x4, x3
    0x10222103c <+60>:  mov    x5, sp
    0x102221040 <+64>:  bl     0x102221088               ; dyldbootstrap::start(macho_header const*, int, char const**, long, macho_header const*, unsigned long*)
->  0x102221044 <+68>:  mov    x16, x0
    0x102221048 <+72>:  ldr    x1, [sp]
    0x10222104c <+76>:  cmp    x1, #0x0
    0x102221050 <+80>:  b.ne   0x10222105c               ; <+92>
    0x102221054 <+84>:  add    sp, x28, #0x8
    0x102221058 <+88>:  br     x16
    0x10222105c <+92>:  mov    x30, x1
    0x102221060 <+96>:  ldr    x0, [x28, #0x8]
    0x102221064 <+100>: add    x1, x28, #0x10
    0x102221068 <+104>: add    x2, x1, x0, lsl #3
    0x10222106c <+108>: add    x2, x2, #0x8
    0x102221070 <+112>: mov    x3, x2
    0x102221074 <+116>: ldr    x4, [x3]
    0x102221078 <+120>: add    x3, x3, #0x8
    0x10222107c <+124>: cmp    x4, #0x0
    0x102221080 <+128>: b.ne   0x102221074               ; <+116>
    0x102221084 <+132>: br     x16

```

</details>


### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
flutter doctor -v                                                                                                                                                                                                                            main
[✓] Flutter (Channel stable, 3.13.9, on macOS 14.1.1 23B2082 darwin-arm64, locale de-DE)
    • Flutter version 3.13.9 on channel stable at /Users/christian/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d211f42860 (3 weeks ago), 2023-10-25 13:42:25 -0700
    • Engine revision 0545f8705d
    • Dart version 3.1.5
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/christian/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (5 available)
    • iPad von Frohe (mobile) • 41a7c175a5965efc7a7eff1941d6161197ebae16 • ios            • iOS 12.5.7 16H81
    • 13mini (mobile)         • 00008110-001A39580C8A801E                • ios            • iOS 17.1 21B5056e
    • iPhone 15 Pro (mobile)  • 8FC568A7-44A2-4A06-A53D-DC2EE9B3FFF4     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)         • macos                                    • darwin-arm64   • macOS 14.1.1 23B2082 darwin-arm64
    • Chrome (web)            • chrome                                   • web-javascript • Google Chrome 119.0.6045.123
    ! Error: Browsing on the local area network for Apple Watch von Christian. Ensure the device is unlocked and discoverable via Bluetooth. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @christian-froh 
I don't have such an iOS version to check this. I noticed you got the crash from XCode. Is there any error from Flutter you can observe? If so, please share the entire output of `flutter run -v`. Thank you!

**Comment by christian-froh** [comment]
There is no Flutter output regarding the crash. However here are the last lines. If you would like to see all, please let me know. I will add it here.

`[        ] (lldb)     connect
[  +20 ms] (lldb)     run
[ +364 ms] success
[        ] (lldb)     safequit
[ +121 ms] Process 817 detached
[  +30 ms] Application launched on the device. Waiting for Dart VM Service url.`

Regarding the key, this is on purpose, so that I dont share my google key here. So with key same behaviour and on other ios versions it works also even if the key is wrong. So all good there.

Ah and regarding the ios version. Its basically the latest version on an Ipad Air which is some years old.

**Comment by huycozy** [comment]
Yes, please share the entire run output in text format that may be helpful. Meanwhile, I will label the issue for other's input.

**Comment by christian-froh** [comment]
Unfortunately I'm getting this error when I try to run it with Flutter:
`Could not build the precompiled application for the device.
Error (Xcode): Target debug_ios_bundle_flutter_assets failed: Exception: Failed to codesign /Users/christian/Desktop/test/build/ios/Debug-iphoneos/App.framework/App with identity 9D506F67B5F729E936D9C886E8F1AF5D443E106A.`

**Comment by huycozy** [comment]
Please share the entire output of `flutter run -v` so that we can verify this properly.

**Comment by christian-froh** [comment]
but unfortunately it doesnt even build.

[testfile.txt](https://github.com/flutter/flutter/files/13348036/testfile.txt)

**Comment by hellohuanlin** [comment]
cc @vashworth

**Comment by vashworth** [comment]
> Unfortunately I'm getting this error when I try to run it with Flutter: `Could not build the precompiled application for the device. Error (Xcode): Target debug_ios_bundle_flutter_assets failed: Exception: Failed to codesign /Users/christian/Desktop/test/build/ios/Debug-iphoneos/App.framework/App with identity 9D506F67B5F729E936D9C886E8F1AF5D443E106A.`

This error is the same as https://github.com/flutter/flutter/issues/130639. Just guessing, but most likely due to Desktop being cloud sync'd (probably iCloud). Flutter doesn't work well in cloud sync'd directories.

The crash still needs to be investigated in this issue.

**Comment by PANGSHENGFENG69** [comment]
Can anyone please help me?
https://github.com/flutter/flutter/issues/138862

**Comment by eldarkk** [comment]
same issue. App crashes on app start ios 12.5.7

# package
google_maps_flutter: ^2.5.3

**Comment by christian-froh** [comment]
it has been fixed with version 2.6.0. thank yoooou
